common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -239,6 +240,16 @@ class HTTPClient(
     private fun Map<String, Any?>.convert(): JSONObject {
         val mapWithoutInnerMaps = mapValues { (_, value) ->
             value.tryCast<Map<String, Any?>>(ifSuccess = { convert() })
```

Comment:
This would need to be removed I think?

Reply:
could you clarify? this was added to fix a separate but similar issue (the one that César posted, https://github.com/RevenueCat/purchases-android/pull/144)